File: Installation/Attributes.vb
Imports System
Namespace DNNtc

#Region " Attributes "
    ''' <summary>
    ''' This class is used to set information about the package
    ''' </summary>
    <AttributeUsage(AttributeTargets.Class, AllowMultiple:=True)> _
    Public Class PackageProperties
        Inherits Attribute
        ' ReSharper disable UnusedParameter.Local
        ''' <summary>
        ''' Creates a attribute with the right properties to create a package. Use on View element
        ''' </summary>
        ''' <param name="name">The package name</param>
        ''' <param name="viewOrder">Ordinal number for sorting packages in the manifest</param>
        ''' <param name="friendlyName">The package friendly name</param>
        ''' <param name="description">The package description</param>
        ''' <param name="iconFile">The package iconfile name</param>
        ''' <param name="ownerName">Name of Owner</param>
        ''' <param name="ownerOrganization">Organization of owner</param>
        ''' <param name="ownerUrl">Url of owner</param>
        ''' <param name="ownerEmail">Email of owner</param>

        Public Sub New(ByVal name As String, ByVal viewOrder As Integer, ByVal friendlyName As String, ByVal description As String, ByVal iconFile As String, ByVal ownerName As String, ByVal ownerOrganization As String, ByVal ownerUrl As String, ByVal ownerEmail As String)
            'Intentially left empty
        End Sub

        ''' <summary>
        ''' Creates a attribute with the right properties to create a package. Use on other elements
        ''' </summary>
        ''' <param name="name">The package name</param>
        Public Sub New(ByVal name As String)
            ' ReSharper restore UnusedParameter.Local
            'Intentially left empty
        End Sub

    End Class

    ''' <summary>
    ''' This class is used to set information about the module
    ''' </summary>
    <AttributeUsage(AttributeTargets.Class, AllowMultiple:=True)> _
    Public Class ModuleProperties
        Inherits Attribute
        ' ReSharper disable UnusedParameter.Local
        ''' <summary>
        ''' Creates a attribute with the right properties to create a module. Use on View element
        ''' </summary>
        ''' <param name="name">The module name.</param>
        ''' <param name="friendlyname">The module friendlyname.</param>
        ''' <param name="defaultCacheTime">the module default cachetime.</param>

        Public Sub New(ByVal name As String, ByVal friendlyname As String, ByVal defaultCacheTime As Integer)
            'Intentially left empty
        End Sub

        ''' <summary>
        ''' Creates a attribute with the right properties to create a module. Use on other elements
        ''' </summary>
        ''' <param name="name">The module name.</param>
        Public Sub New(ByVal name As String)
            ' ReSharper restore UnusedParameter.Local
            'Intentially left empty
        End Sub

    End Class

    ''' <summary>
    ''' This class is used to indicate which UserControls should be in the install package
    ''' </summary>
    Public Class ModuleControlProperties
        Inherits Attribute
        ' ReSharper disable UnusedParameter.Local
        ''' <summary>
        ''' Initializes a new instance of the <see cref="ModuleControlProperties" /> class.
        ''' </summary>
        ''' <param name="key">The key.</param>
        ''' <param name="title">The title.</param>
        ''' <param name="userControlType">Type of the user control.</param>
        ''' <param name="helpUrl">The help URL.</param>
        ''' <param name="supportsPartialRendering">if set to <c>true</c> [supports partial rendering].</param>
        ''' <param name="supportsPopUps">if set to <c>true</c> [supports pop ups].</param>

        Public Sub New(ByVal key As String, ByVal title As String, ByVal userControlType As ControlType, ByVal helpUrl As String, ByVal supportsPartialRendering As Boolean, ByVal supportsPopUps As Boolean)
            ' ReSharper restore UnusedParameter.Local
            'Intentially left empty
        End Sub

    End Class

    ''' <summary>
    ''' Module permission attribute
    ''' </summary>
    <AttributeUsage(AttributeTargets.[Class], AllowMultiple:=True)> _
    Public Class ModulePermission
        Inherits Attribute
        ' ReSharper disable UnusedParameter.Local
        ''' <summary>
        ''' Empty Constructor for Intellisense
        ''' </summary>
        ''' <param name="code"></param>
        ''' <param name="key"></param>
        ''' <param name="name"></param>
        Public Sub New(ByVal code As String, ByVal key As String, ByVal name As String)
            ' ReSharper restore UnusedParameter.Local
            'Intentially left empty
        End Sub
    End Class

    Public Class BusinessControllerClass
        Inherits Attribute
        'Intentially left empty
    End Class

    Public Class UpgradeEventMessage
        Inherits Attribute
        ' ReSharper disable UnusedParameter.Local
        Public Sub New(ByVal versionList As String)
            ' ReSharper restore UnusedParameter.Local
            'Intentially left empty
        End Sub
    End Class

    <AttributeUsage(AttributeTargets.[Class], AllowMultiple:=True)> _
    Public Class ModuleDependencies
        Inherits Attribute
        ' ReSharper disable UnusedParameter.Local
        Public Sub New(ByVal type As ModuleDependency, ByVal value As String)
            ' ReSharper restore UnusedParameter.Local
            'Intentially left empty
        End Sub
    End Class

#End Region

#Region " Enums "

    Public Enum ControlType
        SkinObject
        Anonymous
        View
        Edit
        Admin
        Host
    End Enum

    Public Enum ModuleDependency
        CoreVersion
        Package
        Permission
        Type
    End Enum

#End Region

End Namespace

File: Installation/Attributes.cs
#region Copyright

// 
// DotNetNuke® - http://www.dotnetnuke.com
// Copyright (c) 2002-2018
// by DotNetNuke Corporation
// 
// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
// 
// The above copyright notice and this permission notice shall be included in all copies or substantial portions 
// of the Software.
// 
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
// DEALINGS IN THE SOFTWARE.
//

#endregion

namespace DNNtc
{
    using System;

    #region Attributes

    /// <summary>
    ///     This class is used to indicate which UserControls should be in the install package
    /// </summary>
    public class ModuleControlProperties : Attribute
    {
        /// <summary>
        ///     Creates a attribute with the right properties to create a control.
        /// </summary>
        /// <param name="key">The key.</param>
        /// <param name="title">The title.</param>
        /// <param name="userControlType">type of the user control.</param>
        /// <param name="helpUrl">The help URL.</param>
        /// <param name="supportsPartialRendering">if set to <c>true</c> [supports partial rendering].</param>
        /// <param name="supportsPopUps">if set to <c>true</c> [supports pop ups].</param>
        public ModuleControlProperties(string key, string title, ControlType userControlType, string helpUrl,
                                       bool supportsPartialRendering = false, bool supportsPopUps = false)
        {
            //Intentially left empty
        }
    }

    /// <summary>
    ///     Module permission attribute
    /// </summary>
    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
    public class ModulePermission : Attribute
    {
        /// <summary>
        ///     Empty Constructor for Intellisense
        /// </summary>
        /// <param name="Code"></param>
        /// <param name="Key"></param>
        /// <param name="Name"></param>
        public ModulePermission(string Code, string Key, string Name)
        {
            //Intentially left empty
        }
    }

    public class BusinessControllerClass : Attribute
    {
        //Intentially left empty
    }

    public class UpgradeEventMessage : Attribute
    {
        public UpgradeEventMessage(string VersionList)
        {
            //Intentially left empty
        }
    }

    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
    public class ModuleDependencies : Attribute
    {
        public ModuleDependencies(ModuleDependency Type, string Value)
        {
            //Intentially left empty
        }
    }

    #endregion

    #region Helper Enumerations

    /// <summary>
    ///     Enum for type of component used.
    /// </summary>
    public enum ComponentType
    {
        Script,
        Assembly,
        Module
    }

    /// <summary>
    ///     Enum for the type of controls DotNetNuke is using.
    /// </summary>
    public enum ControlType
    {
        SkinObject,
        Anonymous,
        View,
        Edit,
        Admin,
        Host
    }

    /// <summary>
    ///     The type of dependency for the DotNetNuke Module
    /// </summary>
    public enum ModuleDependency
    {
        CoreVersion,
        Package,
        Permission,
        Type
    }

    #endregion
}